apps/prairielearn/elements/pl-string-input/pl-string-input.py

```diff
@@ -168,7 +168,10 @@ def render(element_html: str, data: pl.QuestionData) -> str:
 
             # If answer is in a format generated by pl.to_json, convert it
             # back to a standard type (otherwise, do nothing)
-            a_sub = pl.from_json(a_sub)
+            #
+            # Some users were putting numbers into the correct answer. For
+            # backwards compatibility, always convert the answer to a string.
+            a_sub = str(pl.from_json(a_sub))
```

Comment:
I reverted this change from #9438, but this is maybe not necessary, things in `submitted_answers` should always be strings I think.

Does anyone know why `pl.from_json(...)` is used here? When would this ever be some value that was produced with `pl.to_json(...)`?

Reply:
I think it's like this because some elements did this for backwards compatibility, and this copied others. I think it maybe mattered for the symbolic input? Either way agree it's probably not necessary.

Reply:
The only situation I can think of is a question that changes from pl-symbolic-input to pl-string-input for the same answers-name or something similar. I'm not concerned if we don't support that case.

That said: I don't think this comment applies here. This is the submitted answer, not the correct answer.

Reply:
You're right, I didn't pay enough attention with my copy/pasting. I'll back these changes out and open another PR to remove the `pl.from_json`.

Reply:
Removed in 7d3c67a.

Reply:
Follow-up PR opened here: https://github.com/PrairieLearn/PrairieLearn/pull/11748